src/main/java/frc/robot/SwerveModule.java

```diff
@@ -21,7 +21,7 @@
 /**
  * Contains all the hardware and controllers for a swerve module.
  */
-public class SwerveModule {
+public class SwerveModule implements AutoCloseable {
```

Comment:
Stuff doesn't need closing...

Reply:
1. Doesn't seem to hurt to close SwerveModules.
2. Probably better to do that if you want to run a variety of unit tests. See: https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/robot-simulation/unit-testing.html
3. No SwerveModules using the ports (e.g., in other unit tests) can be instantiated until you close them.